app/controllers/posts_controller.rb

```diff
@@ -20,7 +20,7 @@ def create
   private
 
   def timeline_posts
-    @timeline_posts ||= Post.all.ordered_by_most_recent.includes(:user)
+    @timeline_posts = Post.where(user: current_user.friends).or(Post.where(user: current_user)).ordered_by_most_recent
```

Comment:
- [ ] where clause can take an array, so we could re-write this as:
```ruby
Post.where(user: current_user.friends + [current_user.id])
```
I also suggest that you move this logic as a method in user model:

```ruby
def timeline_posts
  Post.where(user: friends + [id])
end
```
to make it reusable.